src/dfx/src/lib/extension/manifest/extension.rs

```diff
@@ -14,15 +21,36 @@ pub struct ExtensionManifest {
     pub categories: Vec<String>,
     pub keywords: Option<Vec<String>>,
     pub description: Option<String>,
-    pub commands: JsonValue,
+    pub subcommands: JsonValue, // TODO: awaiting https://dfinity.atlassian.net/browse/SDK-599
     pub dependencies: Option<HashMap<String, String>>,
 }
 
 impl Display for ExtensionManifest {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
-        let Ok(json) = serde_json::to_string_pretty(self) else {
+        let Ok(s) = toml::to_string_pretty(self) else {
             return Err(std::fmt::Error)
         };
-        write!(f, "{}", json)
+        write!(f, "{}", s)
+    }
+}
+
+impl ExtensionManifest {
+    pub fn from_extension_directory(path: PathBuf) -> Result<Self, ExtensionError> {
+        let manifest_path = path.join(MANIFEST_FILE_NAME);
+        if !manifest_path.exists() {
+            return Err(ExtensionError::ExtensionManifestMissing(
+                path.components()
+                    .last()
+                    .unwrap() // safe to unwrap - the `path` parameter is guaranteed not to be root (`/`)
+                    .as_os_str()
+                    .to_string_lossy()
+                    .to_string(),
+            ));
+        }
+        let ext: ExtensionManifestWrapper =
+            toml::from_str(&dfx_core::fs::read_to_string(&manifest_path)?).map_err(|e| {
+                ExtensionError::ExtensionManifestIsNotValid(Box::new(manifest_path), e)
+            })?;
```

Comment:
```
dfx_core::load_toml_file(manifest_path).map_err(ExtensionManifestIsNotValid)?;
```

Reply:
resolving conversation without changes since we're moving to `json`